site/blog/2022-10-28-optimize-s3-endpoints.mdx

```diff
@@ -0,0 +1,126 @@
+---
+slug: optimize-your-s3-endpoints
+title: Optimize your S3 endpoints for cost and efficiency
+authors: [yrobla]
+---
+
+# Optimize your S3 endpoints for cost and efficiency
+
+Are you using S3 buckets as part of your cloud deployments? How are you accessing them?
+
+When not configured properly, access to the S3 buckets may need internet access. One simple but costly way
+is to rely on <a href="https://docs.aws.amazon.com/en_en/vpc/latest/userguide/vpc-nat-gateway.html" target="_blank">NAT gateways</a>.
+
+But there is a more optimal solution, that is creating <a
+href="https://docs.aws.amazon.com/vpc/latest/privatelink/vpc-endpoints-s3.html"
+target="_blank"> gateway or interface VPC endpoints</a> for each region where you have buckets exposed.
+
+When the VPC endpoints are enabled you could access your S3 buckets using this endpoint. This will allow you
+to control your buckets from internal network with the desired security, and without the extra costs of a
+NAT gateway.
+
+### Why use VPC Interface Endpoints
+
+AWS VPC Interface Endpoints are a component of <a href="https://aws.amazon.com/privatelink">PrivateLink</a> infrastructure. AWS PrivateLink provides private connectivity between virtual private clouds (VPCs),
+supported AWS services, and your on-premises networks withoutexposing your traffic to the public internet.
+
+Relying on PrivateLink offers a set of advantages in terms of **security** and **performance**.
+
+**Security: no internet traversal**
+
+When using endpoints, services will be accessed via Amazon's internal networks to route the traffic. This is
+essential for security services or when hosting any content with sensitive data.
```

Comment:
What are "security services" and how is it essential? You could always SSL encrypt your traffic for security reasons.

I think it would be better to point towards "defense in depth" here, that external attackers can't directly get at services that are kept entirely within a private network, which reduces the surface area to especially harden.